packages/interfaces/src/protocols.ts

```diff
@@ -125,117 +125,136 @@ export type Callback<T extends IDecodedMessage> = (
+  UNAVAILABLE = "Service temporarily unavailable",
+  NO_RLN_PROOF = "RLN proof generation failed",
+  NO_PEERS = "No relay peers available"
+}
 
-  /**
-   * General proof generation error message.
-   * nwaku: https://github.com/waku-org/nwaku/blob/c3cb06ac6c03f0f382d3941ea53b330f6a8dd127/waku/waku_rln_relay/group_manager/group_manager_base.nim#L201C19-L201C42
-   */
-  RLN_PROOF_GENERATION = "Proof generation failed",
+// Filter specific errors
+export enum FilterError {
+  // General errors
+  GENERIC_FAIL = "Generic error",
+  DECODE_FAILED = "Failed to decode",
+  NO_PEER_AVAILABLE = "No peer available",
+  NO_STREAM_AVAILABLE = "No stream available",
+  NO_RESPONSE = "No response received",
+  STREAM_ABORTED = "Stream aborted",
 
-  //
-  // RECEIVE ERRORS SECTION
-  //
-  /**
-   * The pubsub topic configured on the decoder does not match the pubsub topic setup on the protocol.
-   * Ensure that the pubsub topic used for decoder creation is the same as the one used for protocol.
-   */
+  // Filter specific errors
+  REMOTE_PEER_REJECTED = "Remote peer rejected",
+  TOPIC_NOT_CONFIGURED = "Topic not configured",
+  SUBSCRIPTION_FAILED = "Subscription failed",
+  UNSUBSCRIBE_FAILED = "Unsubscribe failed",
+  PING_FAILED = "Ping failed",
   TOPIC_DECODER_MISMATCH = "Topic decoder mismatch",
+  INVALID_DECODER_TOPICS = "Invalid decoder topics",
+  SUBSCRIPTION_LIMIT_EXCEEDED = "Subscription limit exceeded",
+  INVALID_CONTENT_TOPIC = "Invalid content topic",
+  PUSH_MESSAGE_FAILED = "Push message failed",
+  EMPTY_MESSAGE = "Empty message received",
+  MISSING_PUBSUB_TOPIC = "Pubsub topic missing from push message"
+}
 
-  /**
-   * The topics passed in the decoders do not match each other, or don't exist at all.
-   * Ensure that all the pubsub topics used in the decoders are valid and match each other.
-   */
-  INVALID_DECODER_TOPICS = "Invalid decoder topics"
+// Protocol-specific failure interfaces
+export interface LightPushFailure {
+  error: LightPushError;
+  peerId?: PeerId;
+  statusCode?: number;
+  statusDesc?: string;
+  /** Protocol version used when the failure occurred (v2, v3, etc.) */
+  protocolVersion?: string;
 }
 
-export interface Failure {
-  error: ProtocolError;
+export interface FilterFailure {
+  error: FilterError;
   peerId?: PeerId;
 }
 
-export type CoreProtocolResult = ThisOrThat<
+// Protocol-specific result types
+export type LightPushCoreResult = ThisOrThat<
   "success",
   PeerId,
   "failure",
-  Failure
+  LightPushFailure
 >;
 
-export type SDKProtocolResult = ThisAndThat<
+export type FilterCoreResult = ThisOrThat<
+  "success",
+  PeerId,
+  "failure",
+  FilterFailure
+>;
+
+export type LightPushSDKResult = ThisAndThat<
+  "successes",
+  PeerId[],
+  "failures",
+  LightPushFailure[]
+> & {
+  /**
+   * Protocol versions used per peer during this send operation
+   * Key: peer ID string, Value: protocol version (e.g., "v2", "v3")
+   */
+  protocolVersions?: Record<string, string>;
```

Comment:
as we discussed offline multiple times - let's not introduce anything in our public SDK with this change

consumer must not be aware what `LightPush` was used

Reply:
removed